File: Adm/Models/src/Combinacoes.java
package models;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.io.FileReader;
import java.io.FileWriter;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

public class Combinacoes extends CRUD<Combinacao> {

    @Override
    public void salvar() {
        try {
            FileWriter writer = new FileWriter("Data/combinacao.json");
            Gson gson = new Gson();
            gson.toJson(objetos, writer); // Serializa os objetos e escreve no arquivo
            writer.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void abrir() {
        // Inicializa a lista objetos para evitar NullPointerException
        if (objetos == null) {
            objetos = new ArrayList<>();  // Se objetos for null, inicializa como lista vazia
        }

        try {
            FileReader reader = new FileReader("Data/combinacao.json");
            Type listType = new TypeToken<List<Combinacao>>(){}.getType();
            // Lê o arquivo JSON e desserializa para a lista de objetos
            List<Combinacao> listacombinacaos = new Gson().fromJson(reader, listType);
            if (listacombinacaos != null) {
                objetos = listacombinacaos; // Se o arquivo contiver dados, atualiza a lista objetos
            }
            reader.close();
        } catch (FileNotFoundException e) {
            // Arquivo não encontrado, não há dados para carregar
            objetos = new ArrayList<>(); // Inicializa a lista vazia
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: Adm/Models/Combinacoes.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.IO;

namespace Adm.Models
{
    public class Combinacoes : CRUD<Combinacao>
    {

        public override void Salvar()
        {
            if (!Directory.Exists("../../Data"))
            {
                Directory.CreateDirectory("../../Data");
            }

            try
            {
                using (StreamWriter arquivo = new StreamWriter("../../Data/combinacao.json"))
                {
                    var dados = new List<Dictionary<string, object>>();
                    foreach (var combinacao in objetos)
                    {
                        dados.Add(combinacao.ToDict());  
                    }

                    string json = JsonConvert.SerializeObject(dados);
                    arquivo.Write(json);
                }
            }
            catch (IOException e)
            {
                Console.WriteLine(e.Message);
            }
        }

        public override void Abrir()
        {
            try
            {
                using (StreamReader reader = new StreamReader("../../Data/combinacao.json"))
                {
                    string json = reader.ReadToEnd();  // Lê o conteúdo do arquivo JSON
                    var listaCombinacoes = JsonConvert.DeserializeObject<List<Dictionary<string, object>>>(json);

                    if (listaCombinacoes != null)
                    {
                        objetos.Clear();
                        foreach (var obj in listaCombinacoes)
                        {
                            // Cria o objeto combinacao a partir do dicionário
                            var combinacao = new Combinacao(
                                Convert.ToInt32(obj["id"]),
                                obj["clima"].ToString(),
                                Convert.ToInt32(obj["id_itens"])
                            );

                            objetos.Add(combinacao);  // Adiciona o combinacao à lista
                        }
                    }
                }
            }
            catch (FileNotFoundException)
            {
                // Não faz nada se o arquivo não for encontrado
            }
            catch (IOException e)
            {
        
            }
        }
    }
}
